# Generated by Django 4.2.7 on 2023-12-07 12:59

from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('core', '0019_remove_order_unique_owner_product_not_canceled_and_more'),
    ]

    operations = [
        migrations.RemoveConstraint(
            model_name='contract',
            name='signed_on_complete',
        ),
        migrations.RemoveConstraint(
            model_name='contract',
            name='reference_datetime_not_both_set',
        ),
        migrations.RemoveConstraint(
            model_name='contract',
            name='signature_backend_reference_must_have_date',
        ),
        migrations.RenameField(
            model_name='contract',
            old_name='signed_on',
            new_name='student_signed_on',
        ),
        migrations.AddConstraint(
            model_name='contract',
            constraint=models.CheckConstraint(check=models.Q(models.Q(('student_signed_on__isnull', False), models.Q(('definition_checksum', ''), _negated=True), models.Q(('definition_checksum__isnull', True), _negated=True), models.Q(('context', None), _negated=True), models.Q(('context', {}), _negated=True)), ('student_signed_on__isnull', True), _connector='OR'), name='student_signed_on_complete', violation_error_message='Make sure to complete all fields before signing contract.'),
        ),
        migrations.AddConstraint(
            model_name='contract',
            constraint=models.CheckConstraint(check=models.Q(('student_signed_on__isnull', False), ('submitted_for_signature_on__isnull', False), _negated=True), name='reference_datetime_not_both_set', violation_error_message='Make sure to not have both datetime fields set simultaneously.'),
        ),
        migrations.AddConstraint(
            model_name='contract',
            constraint=models.CheckConstraint(check=models.Q(('student_signed_on__isnull', False), ('submitted_for_signature_on__isnull', False), ('signature_backend_reference__isnull', True), _connector='OR'), name='signature_backend_reference_must_have_date', violation_error_message='Make sure to have a date attached to the signature backend reference.'),
        ),
    ]
